cmd/cape/cmd/run.go

```diff
@@ -142,23 +144,32 @@ func Run(url string, functionID string, file string, insecure bool) ([]byte, err
 	}
 
 	// TODO: Tuner may want to verify function hash later.
-	res, err := doRun(url, functionID, input, insecure, nil, nil, []string{})
+	res, err := doRun(url, functionID, input, insecure, nil, "", nil, []string{})
 	if err != nil {
 		return nil, fmt.Errorf("error processing data: %w", err)
 	}
 
 	return res, nil
 }
 
-func doRun(url string, functionID string, data []byte, insecure bool, funcHash []byte, keyPolicyHash []byte, pcrSlice []string) ([]byte, error) {
+func doRun(url string, functionID string, data []byte, insecure bool, funcHash []byte, functionToken string, keyPolicyHash []byte, pcrSlice []string) ([]byte, error) { //nolint:gocognit
+	var authToken string
+	var authProtocolType = "cape.runtime"
+
 	endpoint := fmt.Sprintf("%s/v1/run/%s", url, functionID)
 
-	token, err := getAuthToken()
-	if err != nil {
-		return nil, err
+	if functionToken == "" {
+		userToken, err := getAuthToken()
+		if err != nil {
+			return nil, err
+		}
+		authToken = userToken
+	} else {
+		authProtocolType = "cape.function"
```

Comment:
the auth middleware in runtime doesn't let this be anything but cape.runtime. Is the plan to add support after this is merged?

Reply:
We added runtime code to check for this and use the function token context. You already approved that PR.